core/src/main/scala/slamdata/AutoBump.scala

```diff
@@ -0,0 +1,369 @@
+      for {
+        dir <- IO(Files.createTempDirectory("sbt-precog"))
+        dirFile = dir.toFile
+        _ <- run(s"git clone --depth 1 $authenticated ${dirFile.getPath}", log)
+        oldestPullRequest <- getOldestAutoBumpPullRequest
+        (flag, branchName) <- getBranch(oldestPullRequest)
+        _ <- run(s"git checkout $flag $branchName", log, workingDir = Some(dirFile))
+        lines <- run(s"$sbt trickleUpdateDependencies", log, workingDir = Some(dirFile))
+        changes = extractChanges(lines)
+        label = extractLabel(lines)
+        _ <- run(s"$sbt trickleIsUpToDate", log, workingDir = Some(dirFile))
+        updateResult <- run(s"$sbt update", log, merge = true, Some(dirFile)).attempt
+      } yield updateResult.bimap(_ => UpdateError, _ => (dirFile, branchName, oldestPullRequest, changes, label))
+    }
+
+    // TODO: add changes to commit message?
+    def tryCommit(dirFile: File): IO[Either[Warnings, Unit]] = {
+      for {
+        _ <- run(s"git add .", log, merge = true, workingDir = Some(dirFile))
+        result <- run(
+          Seq("git", "commit", "-m", autoBumpCommitTitle(authorRepository)),
+          log,
+          false,
+          Some(dirFile),
+          "GIT_AUTHOR_NAME" -> s"Precog Bot ($authorRepository)",
+          "GIT_AUTHOR_EMAIL" -> "bot@precog.com",
+          "GIT_COMMITTER_NAME" -> s"Precog Bot ($authorRepository)",
+          "GIT_COMMITTER_EMAIL" -> "bot@precog.com").void.attempt
+      } yield result.leftMap(_ => NoChangesError)
+    }
+
+    def tryPush(dirFile: File, branchName: String): IO[Either[Warnings, Unit]] = {
+      run(s"git push origin $branchName", log, merge = true, workingDir = Some(dirFile))
+        .void
+        .attempt
+        .map(_.leftMap(_ => PushError))
+    }
+
+    def ifOldest(pullRequest: PullRequestDraft): IO[Either[Warnings, PullRequestDraft]] = for {
+      _ <- markReady(pullRequest)
+      _ <- IO(log.info(s"Marked $owner/$repoSlug#${pullRequest.number} ready for review"))
+    } yield pullRequest.asRight[Warnings]
+
+    def ifNotOldest(oldest: PullRequestDraft, pullRequest: PullRequestDraft): IO[Either[Warnings, PullRequestDraft]] = for {
+      _ <- close(pullRequest, s"${pullRequest.title} (preceded by #${oldest.number})")
+      _ <- IO(log.info(s"Closed $owner/$repoSlug#${pullRequest.number}, preceded by $owner/$repoSlug#${oldest.number}"))
+      _ <- deleteBranch(pullRequest)
+      _ <- IO(log.info(s"Removed branch ${pullRequest.base.map(_.ref)} from $owner/$repoSlug"))
+    } yield NotOldest(oldest, pullRequest).asLeft[PullRequestDraft]
+
+    // TODO: Update pull request description?
+    def createOrUpdatePullRequest(branchName: String,
+                                  changes: List[String],
+                                  changeLabel: ChangeLabel,
+                                  maybePullRequest: Option[PullRequestDraft]): IO[Either[Warnings, PullRequestDraft]] = {
```

Comment:
Another style example:

```suggestion
    def createOrUpdatePullRequest(
        branchName: String,
        changes: List[String],
        changeLabel: ChangeLabel,
        maybePullRequest: Option[PullRequestDraft])
        : IO[Either[Warnings, PullRequestDraft]] = {
```